src/filter.rs

```diff
@@ -15,167 +15,237 @@
+                    if identifier == "$line" {
+                        return line.contains(&op_value[..]);
+                    } else {
+                        let pval = projection_values.get(&identifier).unwrap();
+                        match pval {
+                            Some(ref s) => {
+                                if s.contains(&op_value) == false {
+                                    return false;
+                                }
+                            }
+                            None => {
+                                return false;
+                            }
+                        };
+                        return true;
+                    }
+                }
+                SQLBinaryOperator::NotLike => {
+                    if identifier != "$line"
+                        && projection_values.contains_key(&identifier[..]) == false
+                    {
+                        return false;
+                    }
+                    // TODO: Optimize this op_value preparation, don't do it in the loop
+                    let op_value = match **right {
+                        ASTNode::SQLIdentifier(ref right_value) => {
+                            // Did they used double quotes for the value?
+                            let mut str_id = right_value.to_string();
+                            if str_id.starts_with("\"") {
+                                str_id = str_id[1..][..str_id.len() - 2].to_string();
+                            }
+                            str_id
+                        }
+                        ASTNode::SQLValue(ref right_value) => match right_value {
+                            sqlparser::sqlast::Value::SingleQuotedString(s) => s.to_string(),
+                            _ => right_value.to_string(),
+                        },
+                        _ => "".to_string(),
+                    };
+                    // TODO: Add support for wildcards ie: LIKE 'server_.domain.com' where _ is a single character wildcard
+                    if identifier == "$line" {
+                        return !line.contains(&op_value[..]);
+                    } else {
+                        let pval = projection_values.get(&identifier).unwrap();
+                        match pval {
+                            Some(ref s) => {
+                                if s.contains(&op_value) == true {
+                                    return false;
+                                }
+                            }
+                            None => {
+                                return false;
+                            }
+                        };
+                        return true;
                     }
                 }
-                _ => {}
+                xop => {
+                    return false;
+                }
             }
         }
-        _ => {}
+        x => {
```

Comment:
Unused binding - AFAICS.

Reply:
same as above, uncommited `println!` updating to `info!`